workflow_imaging/ingest.py

```diff
@@ -1,58 +1,76 @@
+    for session in input_sessions:
         sess_dir = pathlib.Path(session['session_dir'])
-        tiff_filepaths = [fp.as_posix() for fp in sess_dir.glob('*.tif')]
-
-        try:  # attempt to read .tif as a scanimage file
-            loaded_scan = scanreader.read_scan(tiff_filepaths)
-        except Exception as e:
-            print(f'ScanImage loading error: {tiff_filepaths}\n{str(e)}')
-            loaded_scan = None
-
-        if loaded_scan is not None:
-            recording_time = get_scanimage_acq_time(loaded_scan)
-            header = parse_scanimage_header(loaded_scan)
-            session_key = {'subject': session['subject'], 'session_datetime': recording_time}
-            scanner = header['SI_imagingSystem'].strip('\'')
-            if session_key not in Session.proj():
-                scanners.append({'scanner': scanner})
-                sessions.append(session_key)
-                scans.append({**session_key, 'scan_id': 0, 'scanner': scanner})
-
-                session_directories.append({**session_key, 'session_dir': sess_dir.relative_to(root_data_dir).as_posix()})
-                
-                parameter_set = [int(s) for s in session['paramset'].split(' ')]
-                for param in parameter_set:
-                    processing_tasks.append({**session_key, 'scan_id': 0, 'paramset_idx': param, 'task_mode': 'load'})
+        acq_software = None
+        recording_time = None
+        # search for either ScanImage or ScanBox files (in that order)
+        for scan_pattern, scan_type in zip(['*.tif', '*.sbx'], ['ScanImage', 'ScanBox']):
+            scan_filepaths = [fp.as_posix() for fp in sess_dir.glob(scan_pattern)]
+            if len(scan_filepaths):
+                acq_software = scan_type
+                break
+
+        if acq_software == 'ScanImage':
+            import scanreader
+            from elements_imaging.readers import get_scanimage_acq_time, parse_scanimage_header
+            try:  # attempt to read .tif as a scanimage file
+                loaded_scan = scanreader.read_scan(scan_filepaths)
+                recording_time = get_scanimage_acq_time(loaded_scan)
+                header = parse_scanimage_header(loaded_scan)
+                scanner = header['SI_imagingSystem'].strip('\'')
+            except Exception as e:
+                print(f'ScanImage loading error: {scan_filepaths}\n{str(e)}')
+                continue
+        elif acq_software == 'ScanBox':
+            import sbxreader
+            try:  # attempt to load scanbox
+                sbx_fp = pathlib.Path(scan_filepaths[0])
+                sbx_meta = sbxreader.sbx_get_metadata(sbx_fp)
+                recording_time = datetime.fromtimestamp(sbx_fp.stat().st_ctime)  # read from file when scanbox support this
+                scanner = sbx_meta.get('imaging_system', 'ScanBox')
+            except Exception as e:
+                print(f'ScanBox loading error: {scan_filepaths}\n{str(e)}')
+                continue
```

Comment:
Should add an `else` clause to handle the case of `acq_software` being neither `ScanImage` nor `ScanBox`?